pallet-chainlink-feed/src/lib.rs

```diff
@@ -0,0 +1,1367 @@
+
+			Self::deposit_event(RawEvent::OwnerUpdateRequested(feed_id, old_owner, new_owner));
+
+			Ok(().into())
+		}
+
+		/// Accept the transfer of feed ownership.
+		#[weight = T::WeightInfo::accept_ownership()]
+		pub fn accept_ownership(
+			origin,
+			feed_id: T::FeedId,
+		) -> DispatchResultWithPostInfo {
+			let new_owner = ensure_signed(origin)?;
+			let mut feed = Self::feed_config(feed_id).ok_or(Error::<T>::FeedNotFound)?;
+
+			ensure!(feed.pending_owner.filter(|p| p == &new_owner).is_some(), Error::<T>::NotPendingOwner);
+
+			feed.pending_owner = None;
+			feed.owner = new_owner.clone();
+			Feeds::<T>::insert(feed_id, feed);
+
+			Self::deposit_event(RawEvent::OwnerUpdated(feed_id, new_owner));
+
+			Ok(().into())
+		}
+
+		/// Submit a new value to the given feed and round.
+		///
+		/// - Will start a new round if there is no round for the id, yet,
+		///   and a round can be started (at this time by this oracle).
+		/// - Will update the round answer if minimum number of submissions
+		///   has been reached.
+		/// - Records the rewards incurred by the oracle.
+		/// - Removes the details for the previous round if it was superseded.
+		///
+		/// Limited to the oracles of a feed.
+		#[weight = T::WeightInfo::submit_opening_round_answers().max(
+			T::WeightInfo::submit_closing_answer(T::OracleCountLimit::get())
+		)]
+		pub fn submit(
+			origin,
+			feed_id: T::FeedId,
+			round_id: RoundId,
+			submission: T::Value,
+		) -> DispatchResultWithPostInfo {
+			let oracle = ensure_signed(origin)?;
+
+			with_transaction_result(|| -> DispatchResultWithPostInfo {
+				let mut feed = Feed::<T>::load_from(feed_id).ok_or(Error::<T>::FeedNotFound)?;
+				let mut oracle_status = Self::oracle_status(feed_id, &oracle).ok_or(Error::<T>::NotOracle)?;
+				feed.ensure_valid_round(&oracle, round_id)?;
+
+				let (min_val, max_val) = feed.config.submission_value_bounds;
+				ensure!(submission >= min_val, Error::<T>::SubmissionBelowMinimum);
+				ensure!(submission <= max_val, Error::<T>::SubmissionAboveMaximum);
+
+				let new_round_id = feed.reporting_round_id().saturating_add(One::one());
+				let next_eligible_round = oracle_status.last_started_round
+					.unwrap_or(Zero::zero())
+					.checked_add(feed.config.restart_delay).ok_or(Error::<T>::Overflow)?
+					.checked_add(One::one()).ok_or(Error::<T>::Overflow)?;
+				let eligible_to_start = round_id >= next_eligible_round
+					|| oracle_status.last_started_round.is_none();
+
+				// initialize the round if conditions are met
+				if round_id == new_round_id && eligible_to_start {
+					let started_at = feed.initialize_round(new_round_id)?;
+
+					Self::deposit_event(RawEvent::NewRound(feed_id, new_round_id, oracle.clone(), started_at));
+
+					oracle_status.last_started_round = Some(new_round_id);
+				}
+
+				// record submission
+				let mut details = Details::<T>::take(feed_id, round_id).ok_or(Error::<T>::NotAcceptingSubmissions)?;
+				details.submissions.push(submission);
```

Comment:
`max_count != 0` basically checks whether the round details are present in storage (because you get the default value in solidity if the value is not present). We have the `ok_or` for that which checks whether the details are `None`